### PR TITLE
Add fuzzy path search endpoint

### PR DESCRIPTION
**Pull request by ggordonhall**
This PR adds a path search endpoint `search/path` which can be called during the CS flow. It's fuzzy so will return paths that partially match the query too.

This change-set also simplifies `parse_nl` so that it always returns `SemanticQuery`. We weren't using `ParsedQuery::Grep` anywhere meaningful so this should simplify the logic.

### CODE REVIEW

File: server/bleep/src/query/execute.rs [hunk 1]
```diff
@@ -44,6 +45,10 @@ pub struct ApiQuery {
     /// A query written in the bloop query language
     pub q: String,
 
+    /// Optional RepoRef to constrain the search. If not provided, search all repos
+    #[serde(default)]
+    pub repo_ref: Option<RepoRef>,
+
```
Comment (nerdypepper) [inline server/bleep/src/query/execute.rs:51]: don't we have `repo:xxx` for this?

File: server/bleep/src/webserver.rs [hunk 2]
```diff
@@ -54,7 +54,8 @@ pub async fn start(app: Application) -> anyhow::Result<()> {
         .route("/hoverable", get(hoverable::handle))
         .route("/token-info", get(intelligence::handle))
         // misc
-        .route("/search", get(semantic::complex_search))
+        .route("/search/code", get(search::code_search))
+        .route("/search/path", get(search::path_search))
```
Comment (nerdypepper) [inline server/bleep/src/webserver.rs:58]: can we clarify somewhere that these are semantic code search and fuzzy path search? if these endpoints are studio specific, maybe we should use a new route altogether, `studio/xxx`. @calyptobai what do you think?

File: server/bleep/src/webserver/search.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            Err(Error::new(ErrorKind::UpstreamService, "error"))
+        }
+    }
+}
+
+pub(super) async fn path_search(
+    Query(args): Query<ApiQuery>,
+    Extension(indexes): Extension<Arc<Indexes>>,
+) -> impl IntoResponse {
+    match parser::parse_nl(&args.q.clone()) {
```
Comment (nerdypepper) [inline server/bleep/src/webserver/search.rs:40]: we can avoid a `match` here with:

```
let q = parser::parse_nl(..).map_err(|e| {
    // Err arm here
})?;
// Ok arm here
```

**Review by nerdypepper — DISMISSED** [review]

File: server/bleep/src/webserver.rs (hunk 2 above)
Comment (ggordonhall) [reply server/bleep/src/webserver.rs:58]: These endpoints are not strictly studio-specific. They can also be used to debug the `Agent` flow (they simulate the `code` and `path` tools).

**Pull request merged**